<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Best practices on ATAC-seq QC and data analysis • ATACseqQCWorkshop</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/flatly/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Best practices on ATAC-seq QC and data analysis">
<meta property="og:description" content="ATACseqQCWorkshop">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--><!-- Global site tag (gtag.js) - Google Analytics --><script async src="https://www.googletagmanager.com/gtag/js?id=UA-93043521-1"></script><script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'UA-93043521-1');
</script>
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">ATACseqQCWorkshop</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.1.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/ATACseqQC_workshop.html">Best practices on ATAC-seq QC and data analysis</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/haibol2016/ATACseqQCWorkshop">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="ATACseqQC_workshop_files/accessible-code-block-0.0.1/empty-anchor.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Best practices on ATAC-seq QC and data analysis</h1>
                        <h4 class="author">Haibo Liu<a href="#fn1" class="footnote-ref" id="fnref1"><sup>1</sup></a>, Jianhong Ou<a href="#fn2" class="footnote-ref" id="fnref2"><sup>2</sup></a>, Kai Hu<a href="#fn3" class="footnote-ref" id="fnref3"><sup>3</sup></a>, Lihua Julie Zhu<a href="#fn4" class="footnote-ref" id="fnref4"><sup>4</sup></a>
</h4>
            
            <h4 class="date">July 16, 2020</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/haibol2016/ATACseqQCWorkshop/blob/master/vignettes/ATACseqQC_workshop.Rmd"><code>vignettes/ATACseqQC_workshop.Rmd</code></a></small>
      <div class="hidden name"><code>ATACseqQC_workshop.Rmd</code></div>

    </div>

    
    
<div id="overview" class="section level2">
<h2 class="hasAnchor">
<a href="#overview" class="anchor"></a>Overview</h2>
<div id="description" class="section level3">
<h3 class="hasAnchor">
<a href="#description" class="anchor"></a>Description</h3>
<p>In this workshop, we will provide a valuable introduction to the current best practices on ATAC-seq assays, high quality data generation and computational analysis workflow. Then, we will walk the participants through the analysis of an ATAC-seq data set. Detailed tutorials including R scripts will be provided for reproducibility and follow-up practice.</p>
<p>Expectation: After this workshop, participants should be able to apply the learned skills to analyzing their own ATAC-seq data, provide constructive feedback to experimenters who expect to generate high-quality ATAC-seq data, and identify ATAC-seq data of reliable quality for further analysis.</p>
</div>
<div id="pre-requisites" class="section level3">
<h3 class="hasAnchor">
<a href="#pre-requisites" class="anchor"></a>Pre-requisites</h3>
<p>Participants are expected to have basic knowledge as follows:</p>
<ul>
<li>Basic knowledge of R syntax</li>
<li>Some familiarity with the GenomicRanges, BSgenome, GenomicAlignments classes</li>
<li>Familiarity with the SAM file format (<a href="https://samtools.github.io/hts-specs/SAMv1.pdf" class="uri">https://samtools.github.io/hts-specs/SAMv1.pdf</a>)</li>
<li>Basic understanding of how ATAC-seq data is generated is helpful but not required. Please refer to the following reference for detailed information about the ATAC-seq technology.</li>
</ul>
<p>Jason Buenrostro, Beijing Wu, Howard Chang, William Greenleaf. ATAC-seq: A Method for Assaying Chromatin Accessibility Genome-Wide. Curr Protoc Mol Biol. 2015; 109: 21.29.1-21.29.9. <a href="doi:10.1002/0471142727.mb2129s109" class="uri">doi:10.1002/0471142727.mb2129s109</a>.</p>
<p>Please refer to the following resource to preprocess the ATAC-seq data prior to performing quality assessment using the ATACseqQC package.</p>
<p>The Additional File 1 from our publication (Ou et al., 2018; <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5831847/" class="uri">https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5831847/</a>)</p>
</div>
<div id="participation" class="section level3">
<h3 class="hasAnchor">
<a href="#participation" class="anchor"></a>Participation</h3>
<p>After the lecture, participants are expected to follow along the hands-on session. we highly recommend participants bringing your own laptop.</p>
</div>
<div id="r-bioconductor-packages-used" class="section level3">
<h3 class="hasAnchor">
<a href="#r-bioconductor-packages-used" class="anchor"></a><em>R</em> / <em>Bioconductor</em> packages used</h3>
<p>The following R/Bioconductor packages will be explicitly used:</p>
<ul>
<li>ATACseqQC</li>
<li>ChIPpeakAnno</li>
<li>GenomicAlignments</li>
<li>BSgenome.Hsapiens.UCSC.hg19</li>
<li>TxDb.Hsapiens.UCSC.hg19.knownGene</li>
<li>BSgenome.Hsapiens.UCSC.hg38</li>
<li>TxDb.Hsapiens.UCSC.hg38.knownGene</li>
<li>MotifDb</li>
<li>motifStack</li>
</ul>
</div>
<div id="time-outline" class="section level3">
<h3 class="hasAnchor">
<a href="#time-outline" class="anchor"></a>Time outline</h3>
<table class="table">
<thead><tr class="header">
<th>Activity</th>
<th>Time</th>
</tr></thead>
<tbody>
<tr class="odd">
<td>Introduction to ATAC-seq</td>
<td>5m</td>
</tr>
<tr class="even">
<td>Preprocessing of ATAC-seq data</td>
<td>5m</td>
</tr>
<tr class="odd">
<td>ATAC-seq data QC workflow</td>
<td>10m</td>
</tr>
<tr class="even">
<td>Downstream ATAC-seq data analysis</td>
<td>5m</td>
</tr>
<tr class="odd">
<td>Hands on session</td>
<td>30m</td>
</tr>
<tr class="even">
<td>Q &amp; A</td>
<td>5m</td>
</tr>
</tbody>
</table>
</div>
<div id="workshop-goals-and-objectives" class="section level3">
<h3 class="hasAnchor">
<a href="#workshop-goals-and-objectives" class="anchor"></a>Workshop goals and objectives</h3>
<div id="learning-goals" class="section level4">
<h4 class="hasAnchor">
<a href="#learning-goals" class="anchor"></a>Learning goals</h4>
<ul>
<li>Understand how ATAC-seq data are generated<br>
</li>
<li>Learn how to perform comprehensive quality control of ATAC-seq data<br>
</li>
<li>Identify high quality ATAC-seq data for downstream analysis</li>
<li>Identify most likely reasons for ATAC-seq data failing QC<br>
</li>
<li>familiar with ATAC-seq data analysis workflow</li>
</ul>
</div>
<div id="learning-objectives" class="section level4">
<h4 class="hasAnchor">
<a href="#learning-objectives" class="anchor"></a>Learning objectives</h4>
<ul>
<li>Analyze a pre-aligned, excerpted ATAC-seq dataset from the original ATAC-seq publication (Buenrostro et al., 2015) to produce comprehensive insights into the quality of the data</li>
<li>Create a plot showing library fragment size distribution</li>
<li>Create overview plots showing signal distribution around transcription start sites</li>
<li>Create a plot showing CTCF footprints</li>
<li>Evaluate the ATAC-seq data for mitochondrial DNA contamination, duplication rate, background noise level, library complexity and Tn5 transposition optimality</li>
</ul>
</div>
</div>
</div>
<div id="workshop" class="section level2">
<h2 class="hasAnchor">
<a href="#workshop" class="anchor"></a>Workshop</h2>
<div id="introduction-to-the-atac-seq-technology" class="section level3">
<h3 class="hasAnchor">
<a href="#introduction-to-the-atac-seq-technology" class="anchor"></a>Introduction to the ATAC-seq technology</h3>
<hr>
In eukaryotes, nucleosomes are basic DNA packaging units, each of which consists of a nucleosome core particle (NPC), linker DNA and linker histone H1. The NPC is composed <em>ca.</em> 147 bp of DNA wrapped around a histone core octamer, spaced from adjacent NPCs by a linker DNA of <em>ca.</em> 20-90 bp. Linker histones binds additional 20 bp of linker DNA at its entry/exit points to a NPC to stabilize nucleosome conformation and higher-order chromatin assembly. Nucleosomes can be hierarchically assembled into higher-order structures, eventually into chromatin or even chromosomes<span class="citation"><sup>1,2</sup></span> (Figure 1, adapted from <a href="https://www.nature.com/scitable/topicpage/dna-packaging-nucleosomes-and-chromatin-310">Nature Education 1(1):26</a>). In cells, different genomic regions are packaged with different accessibility to transcriptional machinery. Most notably, promoters and enhancers of actively transcribed genes are devoid of histone interaction, which are called “open” chromatin regions (Figure 2, adapted from Wang <em>et al</em><span class="citation"><sup>3</sup></span>). Thus, the interplay between histones and DNA serves as an important layer for controlling gene expression.<span class="citation"><sup>4,5</sup></span> Therefore, it is important to determine the chromatin accessibility to better understand gene expression regulation in cells. <br><div class="figure" style="text-align: center">
<img src="ATACseqQC_workshop_files/figure-html/unnamed-chunk-1-1.png" alt="Figure 1. DNA packaging in eukaryotic cells." width="80%" height="80%"><p class="caption">
Figure 1. DNA packaging in eukaryotic cells.
</p>
</div>
<br><div class="figure" style="text-align: center">
<img src="ATACseqQC_workshop_files/figure-html/unnamed-chunk-2-1.png" alt="Figure 2. Open and closed chromatin." width="80%"><p class="caption">
Figure 2. Open and closed chromatin.
</p>
</div>
<br> In recent years, a few methods have been developed to profile genome-wide chromatin accessibility (Figure 3, adapted from Tsompana and Buck<span class="citation"><sup>6</sup></span>) For review, see Tsompana and Buck.<span class="citation"><sup>6</sup></span> Among these methods, ATAC-seq (an assay for Transposase-Accessible Chromatin using sequencing) is a rapid and sensitive method for profiling chromatin accessibility.<span class="citation"><sup>7</sup></span> Compared to other methods, such as MNase-seq, FAIRE-seq and DNase-seq, ATAC-seq allows comparable or even higher signal-to-noise ratio, but requires much less amount of the biological materials and time to process. Briefly, hyperactive Tn5 transposases preloaded with adapters are first added to simultaneously tag and fragment open chromatin in nuclei (500~50,000) isolated from fresh or cryopreserved tissues/cells at nearly native states, a process called tagmentation. The tagged DNA fragments are then amplified and simultaneously sequencing adapters and indices compatible to Illumina sequencing platforms are added by using PCR with optimized cycles (Figure 4, adapted from Buenrostro <em>et al</em><span class="citation"><sup>7</sup></span>).<br><br><div class="figure" style="text-align: center">
<img src="ATACseqQC_workshop_files/figure-html/unnamed-chunk-3-1.png" alt="Figure 3. Signal features generated by different methods for profiling chromatin accessibility." width="80%"><p class="caption">
Figure 3. Signal features generated by different methods for profiling chromatin accessibility.
</p>
</div>
<p><br></p>
<div class="figure" style="text-align: center">
<img src="ATACseqQC_workshop_files/figure-html/unnamed-chunk-4-1.png" alt="Figure 4. Schematic showing of ATAC-seq library preparation." width="80%" height="80%"><p class="caption">
Figure 4. Schematic showing of ATAC-seq library preparation.
</p>
</div>
<p><br> Since its first debut, ATAC-seq has been used for quite a few purposes:<span class="citation"><sup>7,8</sup></span></p>
<ul>
<li>Profile open chromatin regions<br>
</li>
<li>Infer nucleosome positioning<br>
</li>
<li>Identify transcription factor (TF) footprints<br>
</li>
<li>Infer gene transcriptional regulation<br>
</li>
<li>Determine cell state/identity in combination with single cell technologies</li>
</ul>
The ATAC-seq wet-lab protocol is relatively simple and easy. However, to ensure a successful ATAC-seq experiment, step-by-step quality assurance processes, including both wet lab quality control and <em>in silico</em> quality assessment, are essential (Figure 5, modified from Milani <em>et al</em><span class="citation"><sup>9</sup></span>). In this workshop we will mainly cover the post-alignment QC steps using our ATACseqQC package.<span class="citation"><sup>10</sup></span><br><br><div class="figure" style="text-align: center">
<img src="ATACseqQC_workshop_files/figure-html/unnamed-chunk-5-1.png" alt="Figure 5. Quality control steps in ATAC-seq assays." width="80%"><p class="caption">
Figure 5. Quality control steps in ATAC-seq assays.
</p>
</div>
<p><br></p>
</div>
<div id="preprocessing-of-atac-seq-data" class="section level3">
<h3 class="hasAnchor">
<a href="#preprocessing-of-atac-seq-data" class="anchor"></a>Preprocessing of ATAC-seq data</h3>
<hr>
<p>Before using the ATACSeqQC package to assess the quality of ATAC-seq data, several preprocessing steps are needed as follows. These steps are usually performed in UNIX environments using open source software other than R. Detailed information about how to preprocess ATAC-seq data is provided in <strong>“vignettes/Preprocessing.scripts.for.BioC2020.tutorial.txt”</strong>.</p>
<ul>
<li>Check quality of raw reads using <a href="https://www.bioinformatics.babraham.ac.uk/projects/fastqc/">FASTQC</a>.<br>
</li>
<li>Generate an interactive summary HTML report using MultiQC.<span class="citation"><sup>11</sup></span><br>
</li>
<li>Trim sequencing adapters using Trimmomatic<span class="citation"><sup>12</sup></span> or <a href="http://www.bioinformatics.babraham.ac.uk/projects/trim_galore/">Trim Galore{!}</a>.<br>
</li>
<li>Map the reads to the reference genome of choice using Bowtie/Bowtie2<span class="citation"><sup>13,14</sup></span> or BWA-mem.<span class="citation"><sup>15</sup></span><br>
</li>
<li>Manipulate alignment files using SAMtools<span class="citation"><sup>16</sup></span> or sambamba<span class="citation"><sup>17</sup></span> to get coordinate-sorted BAM files.<br>
</li>
<li>Filter aligned reads
<ul>
<li>Remove reads derived from organelles using SAMtools<span class="citation"><sup>16</sup></span><br>
</li>
<li>Remove duplicates using <a href="https://broadinstitute.github.io/picard/">picard tools</a><br>
</li>
<li>Remove mapping artifacts (fragments &lt; 38 bp<span class="citation"><sup>6</sup></span> and fragments &gt; 2 kb, reads with un-coordinated mapping) using custom scripts</li>
</ul>
</li>
</ul>
</div>
<div id="atacseqqc-workflow" class="section level3">
<h3 class="hasAnchor">
<a href="#atacseqqc-workflow" class="anchor"></a>ATACseqQC workflow</h3>
<hr>
<p>In this workshop we will start with preprocessed alignment files in the BAM formats and use the R/BioConductor package ATACseqQC to perform some representative steps of post-alignment QC.</p>
<div class="sourceCode" id="cb1"><html><body><pre class="r"><span class="co">## prepare the example BAM files for importing</span>
<span class="no">bamFile</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span>(<span class="st">"extdata"</span>, <span class="st">"GL1.bam"</span>,
                        <span class="kw">package</span> <span class="kw">=</span> <span class="st">"ATACseqQC"</span>, <span class="kw">mustWork</span> <span class="kw">=</span> <span class="fl">TRUE</span>)
<span class="no">bamFileLabels</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/grep.html">gsub</a></span>(<span class="st">".bam"</span>, <span class="st">""</span>, <span class="fu"><a href="https://rdrr.io/r/base/basename.html">basename</a></span>(<span class="no">bamFile</span>))</pre></body></html></div>
</div>
<div id="assessing-atac-seq-read-mapping-status" class="section level3">
<h3 class="hasAnchor">
<a href="#assessing-atac-seq-read-mapping-status" class="anchor"></a>Assessing ATAC-seq read mapping status</h3>
<hr>
<p>FASTQC can generate a summary of the quality of raw reads, but it can’t reveal the quality of post-aligned reads or mapping status, such as mapping rate, duplicate rate, genome-wide distribution, mapping quality, and contamination. The <code>bamQC</code> function can serve these needs, which requires sorted BAM files with duplicate reads marked as input.</p>
<div class="sourceCode" id="cb2"><html><body><pre class="r"><span class="co">## bamQC</span>
<span class="fu">bamQC</span>(<span class="no">bamFile</span>, <span class="kw">outPath</span> <span class="kw">=</span> <span class="kw">NULL</span>)</pre></body></html></div>
<pre><code>## $totalQNAMEs
## [1] 44357
## 
## $duplicateRate
## [1] 0
## 
## $mitochondriaRate
## [1] 0
## 
## $properPairRate
## [1] 0.9815136
## 
## $unmappedRate
## [1] 0
## 
## $hasUnmappedMateRate
## [1] 0
## 
## $notPassingQualityControlsRate
## [1] 0
## 
## $nonRedundantFraction
## [1] 0.9398066
## 
## $PCRbottleneckCoefficient_1
## [1] 0.9689694
## 
## $PCRbottleneckCoefficient_2
## [1] 31.22622
## 
## $MAPQ
##    Var1  Freq
## 1    21    44
## 2    22    12
## 3    23   536
## 4    24   606
## 5    25    14
## 6    26    10
## 7    27     4
## 8    30  1682
## 9    31   780
## 10   32   238
## 11   34   244
## 12   35   118
## 13   36    44
## 14   37    62
## 15   38    64
## 16   39    34
## 17   40  2076
## 18   42 82146
## 
## $idxstats
##    seqnames seqlength mapped unmapped
## 1      chr1 249250621  88714        0
## 2      chr2 243199373      0        0
## 3      chr3 198022430      0        0
## 4      chr4 191154276      0        0
## 5      chr5 180915260      0        0
## 6      chr6 171115067      0        0
## 7      chr7 159138663      0        0
## 8      chr8 146364022      0        0
## 9      chr9 141213431      0        0
## 10    chr10 135534747      0        0
## 11    chr11 135006516      0        0
## 12    chr12 133851895      0        0
## 13    chr13 115169878      0        0
## 14    chr14 107349540      0        0
## 15    chr15 102531392      0        0
## 16    chr16  90354753      0        0
## 17    chr17  81195210      0        0
## 18    chr18  78077248      0        0
## 19    chr19  59128983      0        0
## 20    chr20  63025520      0        0
## 21    chr21  48129895      0        0
## 22    chr22  51304566      0        0
## 23     chrX 155270560      0        0
## 24     chrY  59373566      0        0
## 25     chrM     16571      0        0</code></pre>
<div id="assessing-insert-size-distribution" class="section level4">
<h4 class="hasAnchor">
<a href="#assessing-insert-size-distribution" class="anchor"></a>Assessing insert size distribution</h4>
<hr>
It is known that in ATAC-seq experiments, tagmentation of Tn5 transposases produce signature size pattern of fragments derived from nucleosome-free regions, mononucleosome, dinucleosome, trinucleosome and longer oligonucleosome from other open chromatin regions (Figure 6, adapted from Li <em>et al</em><span class="citation"><sup>18</sup></span>) . A typical distribution of insert fragment size is shown in (Figure 7, adapted from Buenrostro <em>et al</em>.<span class="citation"><sup>19</sup></span> The function <code>fragSizeDist</code> in the ATACseqQC package can be used to generate such a distribution plot. Please note the pre-filtered BAM files need to be used to get an unbiased distribution of insert fragment size in the ATAC-seq library.<br><br><div class="figure" style="text-align: center">
<img src="ATACseqQC_workshop_files/figure-html/unnamed-chunk-6-1.png" alt="Figure 6. Fragments generated by Tn5 tagementation in ATAC-seq." width="80%"><p class="caption">
Figure 6. Fragments generated by Tn5 tagementation in ATAC-seq.
</p>
</div>
<br><div class="figure" style="text-align: center">
<img src="ATACseqQC_workshop_files/figure-html/unnamed-chunk-7-1.png" alt="Figure 7. Library insert fragment size distribution in a typical ATAC-seq assay." width="80%"><p class="caption">
Figure 7. Library insert fragment size distribution in a typical ATAC-seq assay.
</p>
</div>
<p><br></p>
<div class="sourceCode" id="cb4"><html><body><pre class="r"><span class="no">fragSize</span> <span class="kw">&lt;-</span> <span class="fu">fragSizeDist</span>(<span class="no">bamFile</span>, <span class="no">bamFileLabels</span>)</pre></body></html></div>
<div class="figure" style="text-align: center">
<img src="ATACseqQC_workshop_files/figure-html/fragSizeDist-1.png" alt="Figure 8. Size distribution of ATAC-seq library insert fragments." width="768"><p class="caption">
Figure 8. Size distribution of ATAC-seq library insert fragments.
</p>
</div>
<p><br></p>
</div>
<div id="shifting-aligned-reads" class="section level4">
<h4 class="hasAnchor">
<a href="#shifting-aligned-reads" class="anchor"></a>Shifting aligned reads</h4>
<hr>
<p>Tagmentation of Tn5 transposases produce 5’ overhang of 9 base long, the coordinates of reads mapping to the positive and negative strands need to be shifted by + 4 and - 5, respectively.<span class="citation"><sup>7,20</sup></span> The functions <code>shiftGAlignmentsList</code> and <code>shiftReads</code> in the ATACseqQC package can be used for this purpose.</p>
<div class="sourceCode" id="cb5"><html><body><pre class="r"><span class="co">## bamfile tags to be read in</span>
<span class="no">possibleTag</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/utils/combn.html">combn</a></span>(<span class="no">LETTERS</span>, <span class="fl">2</span>)
<span class="no">possibleTag</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span>(<span class="no">possibleTag</span>[<span class="fl">1</span>, ], <span class="no">possibleTag</span>[<span class="fl">2</span>, ]),
                 <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span>(<span class="no">possibleTag</span>[<span class="fl">2</span>, ], <span class="no">possibleTag</span>[<span class="fl">1</span>, ]))
<span class="no">bamTop100</span> <span class="kw">&lt;-</span> <span class="fu">scanBam</span>(<span class="fu">BamFile</span>(<span class="no">bamFile</span>, <span class="kw">yieldSize</span> <span class="kw">=</span> <span class="fl">100</span>),
                     <span class="kw">param</span> <span class="kw">=</span> <span class="fu">ScanBamParam</span>(<span class="kw">tag</span> <span class="kw">=</span> <span class="no">possibleTag</span>))<span class="kw">[[</span><span class="fl">1</span>]]$<span class="no">tag</span>
<span class="no">tags</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span>(<span class="no">bamTop100</span>)[<span class="fu"><a href="https://rdrr.io/r/base/lengths.html">lengths</a></span>(<span class="no">bamTop100</span>)<span class="kw">&gt;</span><span class="fl">0</span>]

<span class="co">## files will be output into outPath</span>
<span class="no">outPath</span> <span class="kw">&lt;-</span> <span class="st">"splitBam"</span>
<span class="kw">if</span> (<span class="fu"><a href="https://rdrr.io/r/base/files2.html">dir.exists</a></span>(<span class="no">outPath</span>))
{
  <span class="fu"><a href="https://rdrr.io/r/base/unlink.html">unlink</a></span>(<span class="no">outPath</span>, <span class="kw">recursive</span> <span class="kw">=</span> <span class="fl">TRUE</span>, <span class="kw">force</span> <span class="kw">=</span> <span class="fl">TRUE</span>)
}
<span class="fu"><a href="https://rdrr.io/r/base/files2.html">dir.create</a></span>(<span class="no">outPath</span>)
<span class="co">## shift the coordinates of 5'ends of alignments in the bam file</span>
<span class="no">seqlev</span> <span class="kw">&lt;-</span> <span class="st">"chr1"</span> <span class="co">## subsample data for quick run</span>
<span class="no">which</span> <span class="kw">&lt;-</span> <span class="fu">as</span>(<span class="fu">seqinfo</span>(<span class="no">Hsapiens</span>)[<span class="no">seqlev</span>], <span class="st">"GRanges"</span>)
<span class="no">gal</span> <span class="kw">&lt;-</span> <span class="fu">readBamFile</span>(<span class="no">bamFile</span>, <span class="kw">tag</span><span class="kw">=</span><span class="no">tags</span>, <span class="kw">which</span><span class="kw">=</span><span class="no">which</span>,
                   <span class="kw">asMates</span><span class="kw">=</span><span class="fl">TRUE</span>, <span class="kw">bigFile</span><span class="kw">=</span><span class="fl">TRUE</span>)
<span class="no">shiftedBamFile</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html">file.path</a></span>(<span class="no">outPath</span>, <span class="st">"shifted.bam"</span>)
<span class="no">gal1</span> <span class="kw">&lt;-</span> <span class="fu">shiftGAlignmentsList</span>(<span class="no">gal</span>, <span class="kw">outbam</span><span class="kw">=</span><span class="no">shiftedBamFile</span>)</pre></body></html></div>
</div>
<div id="splitting-bam-files" class="section level4">
<h4 class="hasAnchor">
<a href="#splitting-bam-files" class="anchor"></a>Splitting BAM files</h4>
<hr>
<p>The shifted reads need to be split into different bins, namely bins for reads from nucleosome-free regions, and regions occupied by mononucleosome, dinucleosomes, and trinucleosomes. Shifted reads that do not fit into any of the three bins could be discarded. The function <code>splitGAlignmentsByCut</code> has been implemented to meet this demand.</p>
<div class="sourceCode" id="cb6"><html><body><pre class="r"><span class="co">## run program for chromosome 1 only</span>
<span class="no">txs</span> <span class="kw">&lt;-</span> <span class="fu">transcripts</span>(<span class="no">TxDb.Hsapiens.UCSC.hg19.knownGene</span>)
<span class="no">txs</span> <span class="kw">&lt;-</span> <span class="no">txs</span>[<span class="fu">seqnames</span>(<span class="no">txs</span>) <span class="kw">%in%</span> <span class="st">"chr1"</span>]
<span class="no">genome</span> <span class="kw">&lt;-</span> <span class="no">Hsapiens</span>
<span class="co">## split the reads into bins for reads derived from nucleosome-free regions, mononucleosome, dinucleosome and trinucleosome. And save the binned alignments into bam files.</span>
<span class="no">objs</span> <span class="kw">&lt;-</span> <span class="fu">splitGAlignmentsByCut</span>(<span class="no">gal1</span>, <span class="kw">txs</span><span class="kw">=</span><span class="no">txs</span>, <span class="kw">genome</span><span class="kw">=</span><span class="no">genome</span>, <span class="kw">outPath</span> <span class="kw">=</span> <span class="no">outPath</span>)
<span class="co">## list the files generated by splitGAlignmentsByCut.</span>
<span class="fu"><a href="https://rdrr.io/r/base/list.files.html">dir</a></span>(<span class="no">outPath</span>)</pre></body></html></div>
<pre><code>##  [1] "dinucleosome.bam"       "dinucleosome.bam.bai"   "inter1.bam"            
##  [4] "inter1.bam.bai"         "inter2.bam"             "inter2.bam.bai"        
##  [7] "inter3.bam"             "inter3.bam.bai"         "mononucleosome.bam"    
## [10] "mononucleosome.bam.bai" "NucleosomeFree.bam"     "NucleosomeFree.bam.bai"
## [13] "others.bam"             "others.bam.bai"         "shifted.bam"           
## [16] "shifted.bam.bai"        "trinucleosome.bam"      "trinucleosome.bam.bai"</code></pre>
</div>
<div id="plotting-global-signal-distribution-around-transcriptional-start-sites-tsss" class="section level4">
<h4 class="hasAnchor">
<a href="#plotting-global-signal-distribution-around-transcriptional-start-sites-tsss" class="anchor"></a>Plotting global signal distribution around transcriptional start sites (TSSs)</h4>
<hr>
Previous studies have shown that transcriptionally active elements, such as promoters and enhancers, are defined by short regions of DNA that are devoid of direct histone interactions. These regions of “open” chromatin are usually occupied by transcription factors that facilitate gene transcription. By contrast, the promoters of genes that are not actively expressed in a given cell type exhibit much tighter association with histones, which prevents transcription factors from activating transcription and contributes to gene repression. Typical nucleosome density distributions around TSSs of actively transcribed genes and active enhancers are shown in Figure 9, adapted from Baldi.<span class="citation"><sup>21</sup></span> The function <code>enrichedFragments</code> is for calculating signals around TSSs using the split BAM objects. And the function <code>featureAlignedHeatmap</code> is for generating a heatmap showing signal distribution around TSSs. The function <code>matplot</code> is for summarizing the signal distribution around TSSs using a density plot.<br><br><div class="figure" style="text-align: center">
<img src="ATACseqQC_workshop_files/figure-html/unnamed-chunk-8-1.png" alt="Figure 9. Nucleosome density distribution around TSSs and TF binding sites." width="80%"><p class="caption">
Figure 9. Nucleosome density distribution around TSSs and TF binding sites.
</p>
</div>
<p><br></p>
<div class="sourceCode" id="cb8"><html><body><pre class="r"><span class="no">bamFiles</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html">file.path</a></span>(<span class="no">outPath</span>,
                     <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"NucleosomeFree.bam"</span>,
                     <span class="st">"mononucleosome.bam"</span>,
                     <span class="st">"dinucleosome.bam"</span>,
                     <span class="st">"trinucleosome.bam"</span>))

<span class="no">TSS</span> <span class="kw">&lt;-</span> <span class="fu">promoters</span>(<span class="no">txs</span>, <span class="kw">upstream</span><span class="kw">=</span><span class="fl">0</span>, <span class="kw">downstream</span><span class="kw">=</span><span class="fl">1</span>)
<span class="no">TSS</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/unique.html">unique</a></span>(<span class="no">TSS</span>)
<span class="co">## estimate the library size for normalization</span>
(<span class="no">librarySize</span> <span class="kw">&lt;-</span> <span class="fu">estLibSize</span>(<span class="no">bamFiles</span>))</pre></body></html></div>
<pre><code>## splitBam/NucleosomeFree.bam splitBam/mononucleosome.bam 
##                       17704                        5379 
##   splitBam/dinucleosome.bam  splitBam/trinucleosome.bam 
##                        4907                         923</code></pre>
<div class="sourceCode" id="cb10"><html><body><pre class="r"><span class="co">## calculate the signals around TSSs.</span>
<span class="no">NTILE</span> <span class="kw">&lt;-</span> <span class="fl">101</span>
<span class="no">dws</span> <span class="kw">&lt;-</span> <span class="no">ups</span> <span class="kw">&lt;-</span> <span class="fl">1010</span>
<span class="no">sigs</span> <span class="kw">&lt;-</span> <span class="fu">enrichedFragments</span>(<span class="kw">gal</span><span class="kw">=</span><span class="no">objs</span>[<span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"NucleosomeFree"</span>,
                                     <span class="st">"mononucleosome"</span>,
                                     <span class="st">"dinucleosome"</span>,
                                     <span class="st">"trinucleosome"</span>)],
                          <span class="kw">TSS</span><span class="kw">=</span><span class="no">TSS</span>,
                          <span class="kw">librarySize</span><span class="kw">=</span><span class="no">librarySize</span>,
                          <span class="kw">seqlev</span><span class="kw">=</span><span class="no">seqlev</span>,
                          <span class="kw">TSS.filter</span><span class="kw">=</span><span class="fl">0.5</span>,
                          <span class="kw">n.tile</span> <span class="kw">=</span> <span class="no">NTILE</span>,
                          <span class="kw">upstream</span> <span class="kw">=</span> <span class="no">ups</span>,
                          <span class="kw">downstream</span> <span class="kw">=</span> <span class="no">dws</span>)
<span class="co">## log2 transformed signals</span>
<span class="no">sigs.log2</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html">lapply</a></span>(<span class="no">sigs</span>, <span class="kw">function</span>(<span class="no">.ele</span>) <span class="fu"><a href="https://rdrr.io/r/base/Log.html">log2</a></span>(<span class="no">.ele</span>+<span class="fl">1</span>))
<span class="co">#plot heatmap</span>
<span class="fu">featureAlignedHeatmap</span>(<span class="no">sigs.log2</span>, <span class="fu">reCenterPeaks</span>(<span class="no">TSS</span>, <span class="kw">width</span><span class="kw">=</span><span class="no">ups</span>+<span class="no">dws</span>),
                      <span class="kw">zeroAt</span><span class="kw">=</span><span class="fl">.5</span>, <span class="kw">n.tile</span><span class="kw">=</span><span class="no">NTILE</span>)</pre></body></html></div>
<div class="figure" style="text-align: center">
<img src="ATACseqQC_workshop_files/figure-html/heatmap-1.png" alt="Figure 10. Heatmap showing distributions of reads from different bins around TSSs." width="384"><p class="caption">
Figure 10. Heatmap showing distributions of reads from different bins around TSSs.
</p>
</div>
<div class="sourceCode" id="cb11"><html><body><pre class="r"><span class="co">## get signals normalized for nucleosome-free and nucleosome-bound regions.</span>
<span class="no">out</span> <span class="kw">&lt;-</span> <span class="fu">featureAlignedDistribution</span>(<span class="no">sigs</span>,
                                  <span class="fu">reCenterPeaks</span>(<span class="no">TSS</span>, <span class="kw">width</span><span class="kw">=</span><span class="no">ups</span>+<span class="no">dws</span>),
                                  <span class="kw">zeroAt</span> <span class="kw">=</span> <span class="fl">.5</span>, <span class="kw">n.tile</span> <span class="kw">=</span> <span class="no">NTILE</span>, <span class="kw">type</span> <span class="kw">=</span> <span class="st">"l"</span>,
                                  <span class="kw">ylab</span> <span class="kw">=</span> <span class="st">"Averaged coverage"</span>)</pre></body></html></div>
<div class="sourceCode" id="cb12"><html><body><pre class="r"><span class="co">## rescale the nucleosome-free and nucleosome signals to 0~1</span>
<span class="no">range01</span> <span class="kw">&lt;-</span> <span class="kw">function</span>(<span class="no">x</span>){(<span class="no">x</span>-<span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">min</a></span>(<span class="no">x</span>))/(<span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">max</a></span>(<span class="no">x</span>)-<span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">min</a></span>(<span class="no">x</span>))}
<span class="no">out</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/apply.html">apply</a></span>(<span class="no">out</span>, <span class="fl">2</span>, <span class="no">range01</span>)
<span class="fu"><a href="https://rdrr.io/r/graphics/matplot.html">matplot</a></span>(<span class="no">out</span>, <span class="kw">type</span><span class="kw">=</span><span class="st">"l"</span>, <span class="kw">xaxt</span><span class="kw">=</span><span class="st">"n"</span>,
        <span class="kw">xlab</span><span class="kw">=</span><span class="st">"Position (bp)"</span>,
        <span class="kw">ylab</span><span class="kw">=</span><span class="st">"Fraction of signal"</span>)
<span class="fu"><a href="https://rdrr.io/r/graphics/axis.html">axis</a></span>(<span class="fl">1</span>, <span class="kw">at</span><span class="kw">=</span><span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq</a></span>(<span class="fl">0</span>, <span class="fl">100</span>, <span class="kw">by</span><span class="kw">=</span><span class="fl">10</span>)+<span class="fl">1</span>,
     <span class="kw">labels</span><span class="kw">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"-1K"</span>, <span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq</a></span>(-<span class="fl">800</span>, <span class="fl">800</span>, <span class="kw">by</span><span class="kw">=</span><span class="fl">200</span>), <span class="st">"1K"</span>), <span class="kw">las</span><span class="kw">=</span><span class="fl">2</span>)
<span class="fu"><a href="https://rdrr.io/r/graphics/abline.html">abline</a></span>(<span class="kw">v</span><span class="kw">=</span><span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq</a></span>(<span class="fl">0</span>, <span class="fl">100</span>, <span class="kw">by</span><span class="kw">=</span><span class="fl">10</span>)+<span class="fl">1</span>, <span class="kw">lty</span><span class="kw">=</span><span class="fl">2</span>, <span class="kw">col</span><span class="kw">=</span><span class="st">"gray"</span>)</pre></body></html></div>
<div class="figure" style="text-align: center">
<img src="ATACseqQC_workshop_files/figure-html/unnamed-chunk-10-1.png" alt="Figure 11. Density plot showing distributions of reads from different bins around TSSs. Black line, signal generated from nucleosome-free bin; red line, singal from mononucleosome bin." width="576"><p class="caption">
Figure 11. Density plot showing distributions of reads from different bins around TSSs. Black line, signal generated from nucleosome-free bin; red line, singal from mononucleosome bin.
</p>
</div>
</div>
<div id="streamlining-igv-snapshots-showing-signal-distribution-along-housekeeping-genes" class="section level4">
<h4 class="hasAnchor">
<a href="#streamlining-igv-snapshots-showing-signal-distribution-along-housekeeping-genes" class="anchor"></a>Streamlining IGV snapshots showing signal distribution along housekeeping genes</h4>
<hr>
<p>Housekeeping genes are relatively stably expressed across tissues. Signal enrichment is expected in the regulatory regions of housekeeping genes in successful ATAC-seq experiments, which provides valuable insights into the quality of the ATAC-seq library. We have implemented the function <code>IGVSnapshot</code> is to facilitate automatic visualization of signal distribution along any genomic region of interest. CAUTION: no white spaces are allowed in the full path to BAM files.</p>
<div class="sourceCode" id="cb13"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/r/base/source.html">source</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span>(<span class="st">"extdata"</span>, <span class="st">"IGVSnapshot.R"</span>, <span class="kw">package</span> <span class="kw">=</span> <span class="st">"ATACseqQC"</span>))
<span class="fu">IGVSnapshot</span>(<span class="kw">maxMem</span><span class="kw">=</span><span class="st">"lm"</span>, <span class="kw">genomeBuild</span> <span class="kw">=</span> <span class="st">"hg19"</span>,
            <span class="kw">bamFileFullPathOrURLs</span> <span class="kw">=</span> <span class="no">bamFile</span>,
            <span class="kw">geneNames</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"PQLC2"</span>, <span class="st">"MINOS1"</span>))</pre></body></html></div>
</div>
<div id="assessing-footprints-of-dna-binding-factors" class="section level4">
<h4 class="hasAnchor">
<a href="#assessing-footprints-of-dna-binding-factors" class="anchor"></a>Assessing footprints of DNA-binding factors</h4>
<hr>
<p>Genomic regions bound by transcription factors/insulators are locally protected from Tn5 transposase tagmentation and the pattern of Tn5 transposase cutting sites around TF-binding sites can be used to infer the footprints of TFs. The function <code>factorFootprints</code> is for plotting footprints of DNA-binding factors.</p>
<div class="sourceCode" id="cb14"><html><body><pre class="r"><span class="co">## foot prints</span>
<span class="no">CTCF</span> <span class="kw">&lt;-</span> <span class="fu">query</span>(<span class="no">MotifDb</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"CTCF"</span>))
<span class="no">CTCF</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">as.list</a></span>(<span class="no">CTCF</span>)
<span class="fu"><a href="https://rdrr.io/r/base/print.html">print</a></span>(<span class="no">CTCF</span><span class="kw">[[</span><span class="fl">1</span>]], <span class="kw">digits</span><span class="kw">=</span><span class="fl">2</span>)</pre></body></html></div>
<pre><code>##      1    2    3     4     5     6     7     8     9    10    11    12    13
## A 0.10 0.16 0.30 0.072 0.012 0.786 0.024 0.122 0.914 0.012 0.376 0.022 0.028
## C 0.36 0.21 0.10 0.826 0.966 0.024 0.620 0.494 0.010 0.008 0.010 0.022 0.002
## G 0.12 0.41 0.44 0.050 0.012 0.108 0.336 0.056 0.048 0.976 0.602 0.606 0.962
## T 0.42 0.22 0.16 0.052 0.010 0.082 0.020 0.328 0.028 0.004 0.012 0.350 0.008
##      14    15    16    17   18   19
## A 0.024 0.096 0.424 0.086 0.12 0.34
## C 0.016 0.818 0.024 0.532 0.35 0.26
## G 0.880 0.038 0.522 0.326 0.12 0.32
## T 0.080 0.048 0.030 0.056 0.41 0.08</code></pre>
<div class="sourceCode" id="cb16"><html><body><pre class="r"><span class="no">sigs</span> <span class="kw">&lt;-</span> <span class="fu">factorFootprints</span>(<span class="no">shiftedBamFile</span>, <span class="kw">pfm</span><span class="kw">=</span><span class="no">CTCF</span><span class="kw">[[</span><span class="fl">1</span>]],
                         <span class="kw">genome</span><span class="kw">=</span><span class="no">genome</span>,
                         <span class="kw">min.score</span><span class="kw">=</span><span class="st">"90%"</span>, <span class="kw">seqlev</span><span class="kw">=</span><span class="no">seqlev</span>,
                         <span class="kw">upstream</span><span class="kw">=</span><span class="fl">100</span>, <span class="kw">downstream</span><span class="kw">=</span><span class="fl">100</span>)</pre></body></html></div>
<div class="figure" style="text-align: center">
<img src="ATACseqQC_workshop_files/figure-html/CTCF_footprints-1.png" alt="Figure 12. Aggregated CTCF footprints." width="576"><p class="caption">
Figure 12. Aggregated CTCF footprints.
</p>
</div>
</div>
<div id="assessing-sequencing-depth-and-library-complexity" class="section level4">
<h4 class="hasAnchor">
<a href="#assessing-sequencing-depth-and-library-complexity" class="anchor"></a>Assessing sequencing depth and library complexity</h4>
<hr>
<p>Although all the above assessments can be used to determine the quality of the available ATAC-seq data, they can’t tell whether the sequencing depth is saturated or not, nor whether the library is complex enough for deeper sequencing. The functions <code>saturationPlot</code> and <code>estimateLibComplexity</code> are for these purposes. CAUTION: only BAM files without removing duplicates are informative for estimating library complexity.</p>
<div class="sourceCode" id="cb17"><html><body><pre class="r"><span class="fu">estimateLibComplexity</span>(<span class="fu">readsDupFreq</span>(<span class="no">bamFile</span>))</pre></body></html></div>
<div class="figure" style="text-align: center">
<img src="ATACseqQC_workshop_files/figure-html/libComplexity-1.png" alt="Figure 13. Scatter plot showing estmated ATAC-seq library complexity." width="576"><p class="caption">
Figure 13. Scatter plot showing estmated ATAC-seq library complexity.
</p>
</div>
</div>
<div id="assessing-similarity-of-replicates" class="section level4">
<h4 class="hasAnchor">
<a href="#assessing-similarity-of-replicates" class="anchor"></a>Assessing similarity of replicates</h4>
<hr>
<p>If multiple ATAC-seq assays have been performed, signal correlation among replicates can be checked using the function <code>plotCorrelation</code>.</p>
<div class="sourceCode" id="cb18"><html><body><pre class="r"><span class="no">path</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span>(<span class="st">"extdata"</span>, <span class="kw">package</span><span class="kw">=</span><span class="st">"ATACseqQC"</span>, <span class="kw">mustWork</span><span class="kw">=</span><span class="fl">TRUE</span>)
<span class="no">bamFiles</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.files.html">dir</a></span>(<span class="no">path</span>, <span class="st">"*.bam$"</span>, <span class="kw">full.name</span><span class="kw">=</span><span class="fl">TRUE</span>)
<span class="no">gals</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html">lapply</a></span>(<span class="no">bamFiles</span>, <span class="kw">function</span>(<span class="no">bamfile</span>){
               <span class="fu">readBamFile</span>(<span class="kw">bamFile</span><span class="kw">=</span><span class="no">bamfile</span>, <span class="kw">tag</span><span class="kw">=</span><span class="fu"><a href="https://rdrr.io/r/base/character.html">character</a></span>(<span class="fl">0</span>),
                          <span class="kw">which</span><span class="kw">=</span><span class="fu">GRanges</span>(<span class="st">"chr1"</span>, <span class="fu">IRanges</span>(<span class="fl">1</span>, <span class="fl">1e6</span>)),
                          <span class="kw">asMates</span><span class="kw">=</span><span class="fl">FALSE</span>)
         })
<span class="no">txs</span> <span class="kw">&lt;-</span> <span class="fu">transcripts</span>(<span class="no">TxDb.Hsapiens.UCSC.hg19.knownGene</span>)
<span class="fu">plotCorrelation</span>(<span class="fu">GAlignmentsList</span>(<span class="no">gals</span>), <span class="no">txs</span>, <span class="kw">seqlev</span><span class="kw">=</span><span class="st">"chr1"</span>)</pre></body></html></div>
<div class="figure" style="text-align: center">
<img src="ATACseqQC_workshop_files/figure-html/sampleCorrelation-1.png" alt="Figure 14. Heatmap showing replicate similarity." width="384"><p class="caption">
Figure 14. Heatmap showing replicate similarity.
</p>
</div>
<p><br></p>
</div>
<div id="assessing-other-qc-metrics" class="section level4">
<h4 class="hasAnchor">
<a href="#assessing-other-qc-metrics" class="anchor"></a>Assessing other QC metrics</h4>
<hr>
<p>A few new QC functions have been added to the ATACseqQC package.</p>
<ul>
<li>
<code>distanceDyad</code>: calculate the distance of potential nucleosome dyad and the linear model for V.<br>
</li>
<li>
<code>NFRscore</code>: calculate the ratio of cutting signal immediately adjacent to TSSs and that located to the regions flanking TSSs.<br>
</li>
<li>
<code>PTscore</code>: calculate the ratio of read coverage over promoters to read coverage over transcript body.<br>
</li>
<li>
<code>TSSEscore</code>: calculate aggregated distribution of reads centered on TSSs and that of reads flanking the corresponding TSSs.<br>
</li>
<li>
<code>vPlot</code>: aggregate ATAC-seq Fragment Midpoint vs. Length for a given motif generated over binding sites within the genome.</li>
</ul>
</div>
</div>
<div id="exercises-and-live-demonstration" class="section level3">
<h3 class="hasAnchor">
<a href="#exercises-and-live-demonstration" class="anchor"></a>Exercises and live demonstration</h3>
<hr>
<p>ATAC-seq datasets from two different studies were downloaded from the ENA Sequence Read Archive (SRA). SRR891269 and SRR891270 are ATAC-seq data for two biological replicates of 50K cells from EBV-transformed lymphoblastoid cell line GM12878.<span class="citation"><sup>7</sup></span> SRR5800801 and SRR5800802 are ATAC-seq data for two replicates of 75k cells from a breast cancer cell line T47 (Valles and Izquierd-Bouldstridge, unpublished). The first dataset is of good quality, while the second is from failed ATAC-seq assays. After raw read quality QC using FASTQC, reads were aligned to the human reference genome <strong>hg38</strong> from the <a href="http://hgdownload.soe.ucsc.edu/goldenPath/hg38/bigZips/hg38.fa.gz">UCSC Genome Browser</a> after removing the alternative sequences (chr_alt chromosomes). The resulting BAM files were further preprocessed as above and are available from <a href="https://drive.google.com/drive/folders/1zGvqnXAIWelgVk9Gw1Kx5ySlCg0iz7AJ?usp=sharing">the Google Drive</a>.</p>
<p>Two subsets of filtered BAM files from SRR891270, containing alignments on <strong>chr20</strong> only, are provided in this workshop package for quick in-class practice.</p>
<p>Due to the tight schedule, we will not demonstrate how to perform the QC analyses using the functions <code>saturationPlot</code>,<code>factorFootprints</code>, <code>NFRscore</code>, <code>PTscore</code>, <code>TSSEscore</code>, or <code>vPlot</code>. You may play with the other data and these functions after class. You are encouraged to post your questions at <a href="https://support.bioconductor.org" class="uri">https://support.bioconductor.org</a> with post title “ATACseqQC”.</p>
<p>For your reference, example R scripts for Tasks 1-5 are available at <strong>“inst/vignettes/BioC2020.demo.ATACseqQC.R”</strong>.</p>
<p><br> First, load the required packages.</p>
<div class="sourceCode" id="cb19"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/r/base/message.html">suppressPackageStartupMessages</a></span>({
  <span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="no">ATACseqQC</span>)
  <span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="no">ChIPpeakAnno</span>)
  <span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="no">BSgenome.Hsapiens.UCSC.hg38</span>)
  <span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="no">TxDb.Hsapiens.UCSC.hg38.knownGene</span>)
  <span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="no">MotifDb</span>)
  <span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="no">SRAdb</span>)
  <span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="no">GenomicAlignments</span>)
})</pre></body></html></div>
<p>Task 1. Check read alignment statistics using the function<code>bamQC</code>.<br>
To get the location of the BAM file for Tasks 1 and 2, , please copy and run the following scripts in a R console.</p>
<div class="sourceCode" id="cb20"><html><body><pre class="r"><span class="no">dupBamFile</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span>(<span class="st">"vignettes/extdata"</span>,
                       <span class="st">"SRR891270.chr20.full.bam"</span>,
                       <span class="kw">package</span> <span class="kw">=</span> <span class="st">"ATACseqQCWorkshop"</span>)</pre></body></html></div>
<p>Task 2. Check library complexity for SRR891270 using the function <code>estimateLibComplexity</code>. CAUTION: duplicated should NOT be removed from the input BAM file.</p>
<p>Using the following code to get access to the BAM file for Tasks 2-4.</p>
<div class="sourceCode" id="cb21"><html><body><pre class="r"><span class="no">bamFile</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span>(<span class="st">"vignettes/extdata"</span>,
                       <span class="st">"SRR891270.chr20.rmdup.bam"</span>,
                       <span class="kw">package</span> <span class="kw">=</span> <span class="st">"ATACseqQCWorkshop"</span>)
<span class="no">bamFileLabels</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/grep.html">gsub</a></span>(<span class="st">".rmdup.bam"</span>, <span class="st">""</span>,
                      <span class="fu"><a href="https://rdrr.io/r/base/basename.html">basename</a></span>(<span class="no">bamFile</span>))</pre></body></html></div>
<p>Task 3. Plot the distribution of library insert size using the function <code>fragSizeDist</code>.</p>
<p>Task 4. Plot global signal enrichment around TSSs using the function <code>enrichedFragments</code>. Please shift, and split the BAM file first. Make sure set seqlev as <strong>chr20</strong>.</p>
<p>Task 5. Visualize signal distribution along genomic regions of genes “SPTLC3”, “CRLS1”, and “NELFCD”, using the function <code>IGVSnapshot</code> and the 4 subsets of filtered BAM files.</p>
<p>Download data to a location without white spaces in the full path from the folder <strong>subsetted_filtered_BAM</strong> in the Google Drive, by following <strong><a href="https://drive.google.com/drive/folders/1zGvqnXAIWelgVk9Gw1Kx5ySlCg0iz7AJ?usp=sharing">the shared link</a></strong>.</p>
<div class="sourceCode" id="cb22"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/r/base/source.html">source</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span>(<span class="st">"extdata"</span>, <span class="st">"IGVSnapshot.R"</span>, <span class="kw">package</span> <span class="kw">=</span> <span class="st">"ATACseqQC"</span>))</pre></body></html></div>
</div>
<div id="best-practices-of-atac-seq-and-atac-seq-data-qc" class="section level3">
<h3 class="hasAnchor">
<a href="#best-practices-of-atac-seq-and-atac-seq-data-qc" class="anchor"></a>Best practices of ATAC-seq and ATAC-seq data QC</h3>
<hr>
<p>A few best practices for ATAC-seq assays are suggested as follows:</p>
<ul>
<li>Digest away background DNA (medium/dead cells) using DNase I<span class="citation"><sup>22</sup></span><br>
</li>
<li>Use fresh/cyropreserved cells/tissues to isolate nuclei<span class="citation"><sup>7,9</sup></span><br>
</li>
<li>Reduce mitochondrial/chloroplast DNA contamination as much as possible by using the Omni-ATAC protocol or other methods<span class="citation"><sup>22–25</sup></span><br>
</li>
<li>Optimize the ratio of the amount of Tn5 enzyme to the number of nuclei<br>
</li>
<li>Optimize the number of PCR cycles<span class="citation"><sup>19</sup></span><br>
</li>
<li>Perform Paired-end (PE) sequencing, e.g., 2 x 50 to 100 bp<br>
</li>
<li>Sequence &gt; 50 M PE reads (~200 M for footprinting analysis)<span class="citation"><sup>7</sup></span>
</li>
</ul>
<p>A few best practices for ATAC-seq data analysis are suggested as follows:</p>
<ul>
<li>Perform raw read QC using <a href="https://www.bioinformatics.babraham.ac.uk/projects/fastqc/">FASTQC</a> before alignment</li>
<li>Perform post-alignment QC using ATACSeqQC<span class="citation"><sup>10</sup></span><br>
</li>
<li>Perform peak calling using a peak caller, such as <a href="https://github.com/taoliu/MACS">MACS2</a><span class="citation"><sup>26</sup></span> in narrowdPeak mode with option settings: “shift -s and extend 2s”, <a href="https://github.com/jsh58/Genrich">Genrich</a>, or HMMRATAC.<span class="citation"><sup>27</sup></span>
</li>
<li>Perform post-peak calling QC
<ul>
<li>Annotate peaks and generate peak distribution among genomic features using ChIPpeakAnno<span class="citation"><sup>28</sup></span><br>
</li>
<li>Obtain functions of genes associated to peaks using the Genomic Regions Enrichment of Annotations Tool (GREAT)<span class="citation"><sup>29</sup></span>
</li>
</ul>
</li>
</ul>
</div>
<div id="session-information" class="section level3">
<h3 class="hasAnchor">
<a href="#session-information" class="anchor"></a>Session information</h3>
<hr>
<pre><code>## R version 4.0.0 (2020-04-24)
## Platform: x86_64-pc-linux-gnu (64-bit)
## Running under: Ubuntu 18.04.4 LTS
## 
## Matrix products: default
## BLAS/LAPACK: /usr/lib/x86_64-linux-gnu/libopenblasp-r0.2.20.so
## 
## locale:
##  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
##  [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
##  [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=C             
##  [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
##  [9] LC_ADDRESS=C               LC_TELEPHONE=C            
## [11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       
## 
## attached base packages:
## [1] stats4    parallel  stats     graphics  grDevices utils     datasets 
## [8] methods   base     
## 
## other attached packages:
##  [1] jpeg_0.1-8.1                           
##  [2] png_0.1-7                              
##  [3] GenomicAlignments_1.25.3               
##  [4] Rsamtools_2.5.3                        
##  [5] SummarizedExperiment_1.19.6            
##  [6] DelayedArray_0.15.7                    
##  [7] matrixStats_0.56.0                     
##  [8] Matrix_1.2-18                          
##  [9] SRAdb_1.51.0                           
## [10] RCurl_1.98-1.2                         
## [11] graph_1.67.1                           
## [12] RSQLite_2.2.0                          
## [13] MotifDb_1.31.0                         
## [14] TxDb.Hsapiens.UCSC.hg19.knownGene_3.2.2
## [15] GenomicFeatures_1.41.2                 
## [16] AnnotationDbi_1.51.1                   
## [17] Biobase_2.49.0                         
## [18] BSgenome.Hsapiens.UCSC.hg19_1.4.3      
## [19] BSgenome_1.57.4                        
## [20] rtracklayer_1.49.3                     
## [21] Biostrings_2.57.2                      
## [22] XVector_0.29.3                         
## [23] ChIPpeakAnno_3.23.3                    
## [24] GenomicRanges_1.41.5                   
## [25] GenomeInfoDb_1.25.8                    
## [26] IRanges_2.23.10                        
## [27] ATACseqQC_1.13.3                       
## [28] S4Vectors_0.27.12                      
## [29] BiocGenerics_0.35.4                    
## 
## loaded via a namespace (and not attached):
##   [1] backports_1.1.8               AnnotationHub_2.21.1         
##   [3] BiocFileCache_1.13.0          lazyeval_0.2.2               
##   [5] splines_4.0.0                 BiocParallel_1.23.2          
##   [7] ggplot2_3.3.2                 digest_0.6.25                
##   [9] ensembldb_2.13.1              htmltools_0.5.0              
##  [11] magrittr_1.5                  memoise_1.1.0                
##  [13] grImport2_0.2-0               limma_3.45.9                 
##  [15] readr_1.3.1                   askpass_1.1                  
##  [17] pkgdown_1.5.1                 prettyunits_1.1.1            
##  [19] colorspace_1.4-1              blob_1.2.1                   
##  [21] rappdirs_0.3.1                xfun_0.15                    
##  [23] dplyr_1.0.0                   crayon_1.3.4                 
##  [25] GEOquery_2.57.0               survival_3.1-12              
##  [27] glue_1.4.1                    gtable_0.3.0                 
##  [29] zlibbioc_1.35.0               Rhdf5lib_1.11.3              
##  [31] HDF5Array_1.17.3              scales_1.1.1                 
##  [33] futile.options_1.0.1          DBI_1.1.0                    
##  [35] edgeR_3.31.4                  Rcpp_1.0.5                   
##  [37] xtable_1.8-4                  progress_1.2.2               
##  [39] bit_1.1-15.2                  htmlwidgets_1.5.1            
##  [41] httr_1.4.1                    ellipsis_0.3.1               
##  [43] pkgconfig_2.0.3               XML_3.99-0.4                 
##  [45] dbplyr_1.4.4                  locfit_1.5-9.4               
##  [47] tidyselect_1.1.0              rlang_0.4.7                  
##  [49] later_1.1.0.1                 polynom_1.4-0                
##  [51] munsell_0.5.0                 BiocVersion_3.12.0           
##  [53] tools_4.0.0                   generics_0.0.2               
##  [55] ade4_1.7-15                   evaluate_0.14                
##  [57] stringr_1.4.0                 fastmap_1.0.1                
##  [59] yaml_2.2.1                    knitr_1.29                   
##  [61] bit64_0.9-7.1                 fs_1.4.2                     
##  [63] purrr_0.3.4                   randomForest_4.6-14          
##  [65] AnnotationFilter_1.13.0       splitstackshape_1.4.8        
##  [67] RBGL_1.65.0                   mime_0.9                     
##  [69] formatR_1.7                   xml2_1.3.2                   
##  [71] biomaRt_2.45.2                compiler_4.0.0               
##  [73] curl_4.3                      interactiveDisplayBase_1.27.5
##  [75] preseqR_4.0.0                 tibble_3.0.3                 
##  [77] stringi_1.4.6                 highr_0.8                    
##  [79] futile.logger_1.4.3           desc_1.2.0                   
##  [81] lattice_0.20-41               ProtGenerics_1.21.0          
##  [83] multtest_2.45.0               vctrs_0.3.2                  
##  [85] pillar_1.4.6                  lifecycle_0.2.0              
##  [87] rhdf5filters_1.1.2            BiocManager_1.30.10          
##  [89] data.table_1.12.8             bitops_1.0-6                 
##  [91] httpuv_1.5.4                  R6_2.4.1                     
##  [93] promises_1.1.1                KernSmooth_2.23-16           
##  [95] motifStack_1.33.6             lambda.r_1.2.4               
##  [97] MASS_7.3-51.5                 assertthat_0.2.1             
##  [99] rhdf5_2.33.7                  openssl_1.4.2                
## [101] rprojroot_1.3-2               GenomicScores_2.1.1          
## [103] regioneR_1.21.1               GenomeInfoDbData_1.2.3       
## [105] hms_0.5.3                     VennDiagram_1.6.20           
## [107] grid_4.0.0                    tidyr_1.1.0                  
## [109] rmarkdown_2.3                 base64enc_0.1-3              
## [111] shiny_1.5.0</code></pre>
</div>
<div id="references" class="section level3">
<h3 class="hasAnchor">
<a href="#references" class="anchor"></a>References</h3>
<hr>
<div id="refs" class="references">
<div id="ref-Pederson1986">
<p>1. Pederson, D. S., Thoma, F. &amp; Simpson, R. T. Core particle, fiber, and transcriptionally active chromatin structure. <em>Annual Review of Cell Biology</em> <strong>2,</strong> 117–147 (1986).</p>
</div>
<div id="ref-Bednar2017">
<p>2. Bednar, J. <em>et al.</em> Structure and dynamics of a 197 bp nucleosome in complex with linker histone H1. <em>Molecular Cell</em> <strong>66,</strong> 384–397.e8 (2017).</p>
</div>
<div id="ref-Wang2012">
<p>3. Wang, Y.-M. <em>et al.</em> Correlation between DNase I hypersensitive site distribution and gene expression in HeLa S3 cells. <em>PloS one</em> <strong>7,</strong> e42414 (2012).</p>
</div>
<div id="ref-Gilbert2005">
<p>4. Gilbert, N. &amp; Ramsahoye, B. The relationship between chromatin structure and transcriptional activity in mammalian genomes. <em>Brief Funct Genomic Proteomic</em> <strong>4,</strong> 129–142 (2005).</p>
</div>
<div id="ref-Perino2016">
<p>5. Perino, M. &amp; Veenstra, G. J. C. Chromatin control of developmental dynamics and plasticity. <em>Developmental Cell</em> <strong>38,</strong> 610–620 (2016).</p>
</div>
<div id="ref-Tsompana2014">
<p>6. Tsompana, M. &amp; Buck, M. J. Chromatin accessibility: A window into the genome. <em>Epigenetics Chromatin</em> <strong>7,</strong> 33 (2014).</p>
</div>
<div id="ref-Buenrostro2013">
<p>7. Buenrostro, J. D., Giresi, P. G., Zaba, L. C., Chang, H. Y. &amp; Greenleaf, W. J. Transposition of native chromatin for fast and sensitive epigenomic profiling of open chromatin, DNA-binding proteins and nucleosome position. <em>Nature Methods</em> <strong>10,</strong> 1213–1218 (2013).</p>
</div>
<div id="ref-Buenrostro2015b">
<p>8. Buenrostro, J. D. <em>et al.</em> Single-cell chromatin accessibility reveals principles of regulatory variation. <em>Nature</em> <strong>523,</strong> 486–490 (2015).</p>
</div>
<div id="ref-Milani2016">
<p>9. Milani, P. <em>et al.</em> Cell freezing protocol suitable for atac-seq on motor neurons derived from human induced pluripotent stem cells. <em>Scientific Reports</em> <strong>6,</strong> 25474 (2016).</p>
</div>
<div id="ref-Ou2018">
<p>10. Ou, J. <em>et al.</em> ATACseqQC: A Bioconductor package for post-alignment quality assessment of ATAC-seq data. <em>BMC Genomics</em> <strong>19,</strong> 169 (2018).</p>
</div>
<div id="ref-Ewels2016">
<p>11. Ewels, P., Magnusson, M., Lundin, S. &amp; Käller, M. MultiQC: Summarize analysis results for multiple tools and samples in a single report. <em>Bioinformatics</em> <strong>32,</strong> 3047–3048 (2016).</p>
</div>
<div id="ref-Bolger2014">
<p>12. Bolger, A. M., Lohse, M. &amp; Usadel, B. Trimmomatic: A flexible trimmer for Illumina sequence data. <em>Bioinformatics</em> <strong>30,</strong> 2114–2120 (2014).</p>
</div>
<div id="ref-Langmead2009">
<p>13. Langmead, B., Trapnell, C., Pop, M. &amp; Salzberg, S. L. Ultrafast and memory-efficient alignment of short DNA sequences to the human genome. <em>Genome Biology</em> <strong>10,</strong> R25 (2009).</p>
</div>
<div id="ref-Langmead2012">
<p>14. Langmead, B. &amp; Salzberg, S. L. Fast gapped-read alignment with Bowtie 2. <em>Nature Methods</em> <strong>9,</strong> 357–359 (2012).</p>
</div>
<div id="ref-Li2009a">
<p>15. Li, H. &amp; Durbin, R. Fast and accurate short read alignment with Burrows–Wheeler transform. <em>Bioinformatics</em> <strong>25,</strong> 1754–1760 (2009).</p>
</div>
<div id="ref-Li2009b">
<p>16. Li, H. <em>et al.</em> The Sequence Alignment/Map format and SAMtools. <em>Bioinformatics</em> <strong>25,</strong> 2078–2079 (2009).</p>
</div>
<div id="ref-Tarasov2015">
<p>17. Tarasov, A., Vilella, A. J., Cuppen, E., Nijman, I. J. &amp; Prins, P. Sambamba: Fast processing of NGS alignment formats. <em>Bioinformatics</em> <strong>31,</strong> 2032–2034 (2015).</p>
</div>
<div id="ref-Li2019">
<p>18. Li, Z. <em>et al.</em> Identification of transcription factor binding sites using ATAC-seq. <em>Genome Biology</em> <strong>20,</strong> 45 (2019).</p>
</div>
<div id="ref-Buenrostro2015a">
<p>19. Buenrostro, J. D., Wu, B., Chang, H. Y. &amp; Greenleaf, W. J. ATAC-seq: A method for assaying chromatin accessibility genome-wide. <em>Current protocols in molecular biology</em> <strong>109,</strong> 21.29.1–21.29.9 (2015).</p>
</div>
<div id="ref-Nag1985">
<p>20. Nag, D. K., DasGupta, U., Adelt, G. &amp; Berg, D. E. IS50-mediated inverse transposition: Specificity and precision. <em>Gene</em> <strong>34,</strong> 17–26 (1985).</p>
</div>
<div id="ref-Baldi2018">
<p>21. Baldi, S. Nucleosome positioning and spacing: From genome-wide maps to single arrays. <em>Essays In Biochemistry</em> <strong>63,</strong> 5–14 (2019).</p>
</div>
<div id="ref-Corces2017">
<p>22. Corces, M. R. <em>et al.</em> An improved ATAC-seq protocol reduces background and enables interrogation of frozen tissues. <em>Nature Methods</em> <strong>14,</strong> 959–962 (2017).</p>
</div>
<div id="ref-Lu2016">
<p>23. Lu, Z., Hofmeister, B. T., Vollmers, C., DuBois, R. M. &amp; Schmitz, R. J. Combining ATAC-seq with nuclei sorting for discovery of cis-regulatory regions in plant genomes. <em>Nucleic Acid Research</em> <strong>45,</strong> e41 (2016).</p>
</div>
<div id="ref-Rickner2019">
<p>24. Rickner, H. D., Niu, S.-Y. &amp; Cheng, C. S. ATAC-seq assay with low mitochondrial dna contamination from primary human CD4+ T lymphocytes. <em>JoVE</em> e59120 (2019). doi:<a href="https://doi.org/10.3791/59120">10.3791/59120</a></p>
</div>
<div id="ref-Montefiori2017">
<p>25. Montefiori, L. <em>et al.</em> Reducing mitochondrial reads in ATAC-seq using CRISPR/Cas9. <em>Scientific Reports</em> <strong>7,</strong> 2451 (2017).</p>
</div>
<div id="ref-Zhang2008">
<p>26. Zhang, Y. <em>et al.</em> Model-based Analysis of ChIP-Seq (MACS). <em>Genome Biology</em> <strong>9,</strong> R137 (2008).</p>
</div>
<div id="ref-Tarbell2019">
<p>27. Tarbell, E. D. &amp; Liu, T. HMMRATAC: A hidden markov modeler for atac-seq. <em>bioRxiv</em> (2019). doi:<a href="https://doi.org/10.1101/306621">10.1101/306621</a></p>
</div>
<div id="ref-Zhu2010">
<p>28. Zhu, L. J. <em>et al.</em> ChIPpeakAnno: A Bioconductor package to annotate ChIP-seq and ChIP-chip data. <em>BMC Bioinformatics</em> <strong>11,</strong> 237 (2010).</p>
</div>
<div id="ref-McLean2010">
<p>29. McLean, C. Y. <em>et al.</em> GREAT improves functional interpretation of cis-regulatory regions. <em>Nature Biotechnology</em> <strong>28,</strong> 495–501 (2010).</p>
</div>
</div>
</div>
</div>
<div class="footnotes">
<hr>
<ol>
<li id="fn1"><p>Department of Molecular, Cell, and Cancer Biology, University of Massachusetts Medical School, 364 Plantation Street, Worcester, MA, 01605, USA. <a href="mailto:haibo.liu@umassmed.edu" class="email">haibo.liu@umassmed.edu</a><a href="#fnref1" class="footnote-back">↩︎</a></p></li>
<li id="fn2"><p>Regeneration NEXT, Duke University School of Medicine, Duke University, Durham, NC, 27701, USA. <a href="mailto:Jianhong.ou@duke.edu" class="email">Jianhong.ou@duke.edu</a><a href="#fnref2" class="footnote-back">↩︎</a></p></li>
<li id="fn3"><p>Department of Molecular, Cell, and Cancer Biology, University of Massachusetts Medical School, 364 Plantation Street, Worcester, MA, 01605, USA. <a href="mailto:Kai.Hu@umassmed.edu" class="email">Kai.Hu@umassmed.edu</a><a href="#fnref3" class="footnote-back">↩︎</a></p></li>
<li id="fn4"><p> Corresponding author, Department of Molecular, Cell and Cancer Biology, Program in Molecular Medicine, Program in Bioinformatics and Integrative Biology, UMass Medical School, Worcester, MA, 01655, USA. <a href="mailto:Julie.Zhu@umassmed.edu" class="email">Julie.Zhu@umassmed.edu</a><a href="#fnref4" class="footnote-back">↩︎</a></p></li>
</ol>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by Haibo Liu, Jianhong Ou, Kai Hu, Liuhua Zhu.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.5.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
